File: Neural-Networks/create_MEGA_THETA.m
% initialise MEGA_THETA, the matrix determining the neural network
 
function [MEGA_THETA] = create_MEGA_THETA(xi)
 
L = length(xi);
sum_xi = sum(xi);
rows = sum_xi - xi(1);
cols = sum_xi - xi(L) + L - 1;      %  include base unit for each layer
MEGA_THETA = eye(rows,cols);        %  put some ones in
 
end

File: Neural-Networks/create_MEGA_THETA.py
# initialise MEGA_THETA, the matrix determining the neural network

from numpy import *

def create_MEGA_THETA(xi):

        L = size(xi)
        sum_xi = sum(xi)
        rows = sum_xi - xi[0]
        cols = sum_xi - xi[-1] + L - 1      #  include base unit for each layer
        MEGA_THETA = random.normal(0,1,rows*cols).reshape((rows,cols))
        return MEGA_THETA
